posts/setting-up-a-shared-library-and-seed-job-in-jenkins-2.md

```diff
@@ -0,0 +1,189 @@
+2. Configure `seed.groovy` to create a Pipeline Job and Multibranch Pipeline Job per service.
+3. Create `jenkinsJob.groovy` to be used by our services as the entry point into our Shared Library.
+4. Set up a [`Jenkinsfile`](https://jenkins.io/doc/book/pipeline/jenkinsfile/) in `poc-micro` to use `microservice-pipelines` Shared Library.
+
+## Goal 1
+### Configure default Shared Library setup for Jenkins
+Since we will be using a Shared library, Jenkins needs to be set up to use our Shared Library.
+
+   1. Navigate to `Dashboard` > select `Manage Jenkins` > select `Configure System` > scroll down to `Global Pipeline Libraries` > select `Add`
+   2. Enter `microservice-pipelines` in the `Name` field
+   3. Ented `master` in `Default Version`
+      * This tells jenkins which branch of our Shared Library we plan to use by default.
+   4. Under `Source Code Management`, select `Git` 
+   5. In the `Project Repository` field, enter `https://github.com/kcrane3576/microservice-pipelines` > select `Save`.
+   ![jenkins shared library configuration](hhttps://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-config-2.png)
+
+## Goal 2
+We are going to modify `seed.groovy` to build a Pipeline job and Multibranch Pipeline job for all services we oboard. 
+
+#### Update `seedJob` to use a `part2` branch we will create in `microservice-pipelines`
+1. Navitate to `Dashboard` > select `seedJob` > select `Configure`.
+2. Under `Source Code Management`, change the `Branch Specifier` to `*/part2`.
+
+### Updating `microservice-pipelines` to build our `pipelineJob` and `multibranchPipelineJob`
+We are going to leave the `master` branch of `microservice-pipelines` alone to ensure it works with Part 1 of this series. In order for us to do this, we will introduce the changes to the `seed.groovy` job on branch `part2` of [`microservice-pipelines`](https://github.com/kcrane3576/microservice-pipelines).
+
+#### Adding `pipelineJob` and `multibranchPipelineJob` to `seed.groovy`
+1. Create a new branch `part2` in `microservice-pipelines`.
+2. In the `part2` branch, remove the original code in `seed.groovy` and paste in the below code.
+   * For a better understanding of the `pipelineJob` and `multibranchPipelineJob`, make sure to go back and check the [Jenkins Job DSL API](https://jenkinsci.github.io/job-dsl-plugin/#).
+```groovy
+def createPipelineJob(jobName, repoUrl) {
+    pipelineJob(jobName) {
+        definition {
+            cpsScm {
+                scm {
+                    git {
+                        remote {
+                            url(repoUrl)
+                        }
+                        branches('master')
+                        extensions {
+                            cleanBeforeCheckout()
+                        }
+                    }
+                }
+                scriptPath("Jenkinsfile")
+            }
+        }
+    }
+}
+
+def createMultibranchPipelineJob(jobName, repoUrl) {
```

Comment:
same, I would name the job by its purpose

Reply:
I renamed the job and updated the code in the blog on the latest version